nodes/default_node.go

```diff
@@ -133,6 +133,9 @@ func (d *DefaultNode) GetContainers(ctx context.Context) ([]runtime.GenericConta
 	if err != nil {
 		return nil, err
 	}
+	if len(cnts) == 0 {
+		return nil, fmt.Errorf("No containers found for node %q", d.GetContainerName())
+	}
```

Comment:
let's make an error type for that and wrap it with `%w` so that the caller can do error.Is() and check if the error is indeed about no containers found to make informed decision about how to handle that

Reply:
Totally agree, appeared a little to late to me.

Reply:
what I meant was

```go
// ErrContainersNotFound is returned when no containers are found.
var ErrContainersNotFound = errors.New("containers not found")

return nil, fmt.Errorf("Node: %s. %w", d.GetContainerName(), ErrContainersNotFound)
```